nixos/modules/virtualisation/docker.nix

```diff
@@ -155,19 +155,17 @@ in
       users.groups.docker.gid = config.ids.gids.docker;
       systemd.packages = [ cfg.package ];
 
-      # TODO: remove once docker 20.10 is released
-      systemd.enableUnifiedCgroupHierarchy = false;
-
       systemd.services.docker = {
         wantedBy = optional cfg.enableOnBoot "multi-user.target";
         environment = proxy_env;
         serviceConfig = {
+          Type = "notify";
           ExecStart = [
             ""
             ''
               ${cfg.package}/bin/dockerd \
                 --group=docker \
-                --host=fd:// \
+                --host=unix:// \
```

Comment:
Why did you need to change this?
Socket activation is still supported, judging from the [docs](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option).

Reply:
A leftover which shouldn't ever got here.

Though I had a very odd issue when I used `fd://` - Docker kept complaining about being unable to find sockets and if I'm even launching it with systemd. Now the issue appears to be vanished 🤔

Reply:
> complaining

Was it a false warning from dockerd or an actual error, like a docker client that can't connect? If it's a warning, that seems acceptable, because then we can revert https://github.com/NixOS/nixpkgs/pull/109418#event-4209070612 and have a reliable socket activation.

Reply:
No, that was an error from `dockerd`, causing daemon not to start.

Reply:
Unable to reproduce the issue on fresh NixOS installation. This change should be definitely reverted.